Update README.md test case and dev jinja check

loudambiance opened the pull request:
Added a check for the dev version of jinja and updated the README.md test.

Commit:
Add link to Jinja2 documentation to help text.

Also changed help text printing to use a formated string to pull in
version from directly from Jinja library.

Commit:
Update README.md to have current help/usage text.

This uses the most recent version of Jinja. I am unsure how to make this
test dynamically pass based on version, without rewritting the test.

loudambiance committed:
Change README test to compare to expected output, instead tpl output.
Updated README to use latest documentation link.
Updated tpl to check for dev version of jinja
and change link to latest documentation.

loudambiance committed:
Fix Merge

loudambiance committed:
Merge in master changes

loudambiance committed:
Change lastest to dev in url per M3t0r's request

M3t0r commented on tests/cli/test_help_and_usage.py:
```diff
@@ -35,6 +35,19 @@ def test_version_on_version(cli):
 
 
 def test_current_help_in_readme(cli):
-    p = cli("--help")
```
That was what I wanted to avoid. If later we change the text in the help message we will have to change it in two places. I would move this test into `tests/test_interface.py` or similar and test the output of `print_help()` with https://docs.pytest.org/en/latest/capture.html?highlight=stdout#accessing-captured-output-from-a-test-function instead of using `cli("--help")`.

loudambiance replied:
I'll take another look at this. I misunderstood, sorry.

M3t0r committed:
Transform test_current_help_in_readme into a unittest

This allows us to mock the Jinja version with the string "latest", allowing
us to have the link to the latest documentation in the README, but still
offer the correct documentation in a command line invocation while
maintaining only one source of truth without duplicating the help text in
multiple places.

M3t0r committed:
Add explanation to --help message

Most tools I checked stick to max 80 chars for their messages even if they
could check the column count of the terminal. So I refrained from having
the text be reflown by the terminal and fixed it to 80 columns.

loudambiance committed:
Merge pull request #1 from M3t0r/improve-readme-help-text-test

Improve test_current_help_in_readme

M3t0r commented:
Thank you for your contributions 😊
